service/worker/scheduler/workflow.go

```diff
@@ -1002,6 +1004,39 @@ func (s *scheduler) getListInfo(inWorkflowContext bool) *schedpb.ScheduleListInf
 	}
 }
 
+func (s *scheduler) updateCustomSearchAttributes(searchAttributes *commonpb.SearchAttributes) {
+	if searchAttributes == nil {
+		return
+	}
+	upsertMap := map[string]any{}
+	for key, valuePayload := range searchAttributes.GetIndexedFields() {
```

Comment:
factor out this code so it works for memo also?

Reply:
Memo field names are different. We can't refactor this.